pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -317,6 +318,24 @@ func (e *Controller) onServiceDelete(obj interface{}) {
 	e.queue.Add(key)
 }
 
+func (e *Controller) onEndpointsUpdate(old, cur interface{}) {
+	newEps := cur.(*v1.Endpoints)
+	oldEps := old.(*v1.Endpoints)
+	if newEps == nil || oldEps == nil {
+		utilruntime.HandleError(fmt.Errorf("Invalid Endpoint provided to onEndpointsUpdate()"))
+		return
+	}
+	key, err := controller.KeyFunc(newEps)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Couldn't get key for object %+v: %v", newEps, err))
+		return
+	}
+	// only process the endpoint update if it has been tracked by this controller
```

Comment:
This is pretty hostile, if I understand it - what about Services with no selector?

Reply:
I mis-read it.  Sorry.  

It seems plausible to re-run Endpoint sync if it changes.  Does this trigger a loop?

@robscott - you know this code better than me now.

Reply:
@liggitt is there any APImachinery thing that allows a controller to create an update, in this case, update an Endpoint, and not process that update on the event handler? i.e.
```
func (e *Controller) onEndpointsUpdate(old, cur interface{}) {
    // enqueue only if we didn't make the update

}
```

Reply:
> @liggitt is there any APImachinery thing that allows a controller to create an update, in this case, update an Endpoint, and not process that update on the event handler? i.e.
> 
> ```
> func (e *Controller) onEndpointsUpdate(old, cur interface{}) {
>     // enqueue only if we didn't make the update
> 
> }
> ```

not really... the normal approach is to inspect the content of the object in the update event, and skip a sync if the current content is acceptable, or if a write attempt gets a conflict error (indicates the observed object is stale)